workflowparsers/quantum_espresso_bands/parser.py

```diff
@@ -0,0 +1,570 @@
+                            dtype=float,
+                        ),
+                        Quantity(
+                            'mult',
+                            r'eV\s*(\d+)\s*-->',
+                            repeats=True,
+                            dtype=int,
+                        ),
+                    ],
+                ),
+            ),
+            Quantity(
+                'failed_symmetry',
+                r'(zone border point and non-symmorphic group)',
+                repeats=True,
+                dtype=str,
+                default='',
+            ),
+        ]
+
+    @staticmethod
+    def scan_dir_for_files(directory: str, pattern: str = '*.out') -> list[str]:
+        """Find all files matching the pattern in the given directory."""
+        return glob.glob(os.path.join(directory, pattern))
+
+    @staticmethod
+    def points_to_segments(kpoints: list, symmetries: list) -> list[list[KPoint]]:
+        """Split the k-points by segment based on differing symmetry group."""
+
+        def shift_window(window: tuple, elem) -> tuple:
+            return window[1:] + (elem,)
+
+        previous_kpoint: Optional[KPoint] = None
+        symmetry_window: tuple[Optional[str], Optional[str], Optional[str]] = (
+            None,
+        ) * 3
+
+        segments: list[list[KPoint]] = [[]]  # segments are lists of k-point paths
+        for kpoint, symmetry in zip(kpoints, symmetries):
+            symmetry_window = shift_window(symmetry_window, symmetry)
+            # case enumeration for `symmetry_window`:
+            # 1. (None, None, None) -> not possible
+            # 2. (None, None, X) -> first step (add to initial bucket)
+            # 3. (X, None, None) -> not possible
+            # 4. (X, Y, None) -> end reached
+            # 5. (None, X, Y) -> add to initial bucket
+            # 6. (X, X, Y) -> add Y to latest bucket
+            # 7. (X, Y, Y) -> add Y to latest bucket
+            # 8. (X, Y, X) -> add Y and X a new, latest bucket
+            # 9. (X, Y, Z) -> add Y and Z a new, latest bucket
+            if (None not in symmetry_window) and all(
+                [symmetry_window[i] != symmetry_window[i + 1] for i in range(2)]
+            ):
+                segments.append([previous_kpoint])
+            segments[-1].append(kpoint)
+            previous_kpoint = kpoint
+        return segments
+
+    @staticmethod
+    def apply_multiplicity(
+        energies: list[list[float]], mults: list[list[int]]
+    ) -> list[list[float]]:
+        """Apply band multiplicity to energies."""
+        return [
+            [e for e, m in zip(energy, mult) for _ in range(m)]
+            for energy, mult in zip(energies, mults)
+        ]
+
+
+class QuantumEspressoBandsParser:
+    """Parser for Quantum ESPRESSO BANDS calculations."""
+
+    def __init__(self):
+        self.bands_parser = BandsFileParser()
+        self.pwscf_parser = MainfileParser()
+
+    def _find_files(self) -> Optional[str]:
```

Comment:
you probably do need to scan the directory for the pwscf files.

Reply:
This is done by `_find_files`, on the line `out_files = self.bands_parser.scan_dir_for_files(self.maindir)`.
Pls let me know whether this resolves your comment.

Reply:
sorry there should be a NOT there.